transparentpath/gcsutils/transparentpath.py

```diff
@@ -1693,6 +1698,39 @@ def cast_iterable(self, iter_: Iterable):
             to_ret = [self.cast_iterable(item) for item in iter_]
             return to_ret
 
+    def caching_saver(self, file):
+        """
+        save fetched data in tmp file or dict, if not exceeding caching_max_memory else remove oldest data
+        """
+        if self.enable_caching:
+            used_memory = 0
+            if self.caching == "ram":
+                if sys.getsizeof(file) > self.caching_max_memory:
+                    raise BufferError("TOO FAT FILE")
```

Comment:
Maybe not raise but warn ? and add a bit more information (like, the path itself, that we are trying to cache a file, etc...)

Separate the big "if" in two methods